tests/test_factory_errors.py

```
@@ -0,0 +1,33 @@
+def test_postgresql_async_raises_on_use_without_pytest_asyncio() -> None:
+    """When pytest-asyncio is absent, the registered stub is synchronous and raises ImportError.
+
+    A synchronous stub avoids the "coroutine was never awaited" warning that would
+    result from registering an async def with plain pytest.fixture.
+    """
+    with patch("pytest_postgresql.factories.client.pytest_asyncio", None):
+        fixture_func = postgresql_async("some_proc_fixture")
+        # pytest 8+ wraps fixtures to prevent direct calls; unwrap first.
+        raw_func = getattr(fixture_func, "__wrapped__", fixture_func)
+        assert not hasattr(raw_func, "__await__"), "stub must be a sync function, not a coroutine"
+        with pytest.raises(ImportError, match="pytest-asyncio >= 0.24"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)